src/models/transfer/facade.js

```diff
@@ -440,32 +436,73 @@ const saveTransferPrepared = async (payload, stateReason = null, hasPassedValida
     }
 
     const knex = await Db.getKnex()
-    return await knex.transaction(async (trx) => {
+    if (hasPassedValidation) {
+      return await knex.transaction(async (trx) => {
+        try {
+          await knex('transfer').transacting(trx).insert(transferRecord)
+          await knex('transferParticipant').transacting(trx).insert(payerTransferParticipantRecord)
+          await knex('transferParticipant').transacting(trx).insert(payeeTransferParticipantRecord)
+          payerTransferParticipantRecord.name = payload.payerFsp
+          payeeTransferParticipantRecord.name = payload.payeeFsp
+          let transferExtensionsRecordList = []
+          if (payload.extensionList && payload.extensionList.extension) {
+            transferExtensionsRecordList = payload.extensionList.extension.map(ext => {
+              return {
+                transferId: payload.transferId,
+                key: ext.key,
+                value: ext.value
+              }
+            })
+            await knex.batchInsert('transferExtension', transferExtensionsRecordList).transacting(trx)
+          }
+          await knex('ilpPacket').transacting(trx).insert(ilpPacketRecord)
+          await knex('transferStateChange').transacting(trx).insert(transferStateChangeRecord)
+          await trx.commit
+        } catch (err) {
+          await trx.rollback
+          throw err
+        }
+      })
+    } else {
+      await knex('transfer').insert(transferRecord)
```

Comment:
Why would these inserts not be in a transaction? I don't see the reason to only insert certain entries in the database.

Reply:
This is a DA decision - to store what we are able to insert.